### PR TITLE
CMake refactoring

### PR DESCRIPTION
**Pull request by Mon1Amour4**
Move Windows implementation to a dedicated WindowsDep static library using Cmake INTERFACE

### CODE REVIEW

File: Engine/CMakeLists.txt [hunk 1]
```diff
@@ -5,6 +5,7 @@ set(FUEGO_DYNAMIC_LIB
     CACHE BOOL "Build dynamic library (ON/OFF)")
 
 include(${CMAKE_CURRENT_SOURCE_DIR}/ImGui.cmake)
+include(${CMAKE_CURRENT_SOURCE_DIR}/WindowsDep.cmake)
```
Comment (LtSnail) [inline Engine/CMakeLists.txt:8]: Please, create a CMakeLists.txt

File: Engine/CMakeLists.txt [hunk 2]
```diff
@@ -32,6 +33,9 @@ set(SOURCES Fuego/Application.cpp Fuego/Log.cpp Fuego/Layer.cpp
             Fuego/LayerStack.cpp Fuego/ImGui/ImGuiLayer.cpp)
 
 set(DEPENDENCIES ImGui)
+if(WIN32)
```
Comment (LtSnail) [inline Engine/CMakeLists.txt:36]: It would be good if we don't have conditions here. You can add a global if which encloses everything in your Windows CMakeLists.txt

File: Engine/CMakeLists.txt [hunk 3]
```diff
@@ -71,35 +75,14 @@ if(APPLE)
...
-  endif()
 endif()
 
 if(FUEGO_DYNAMIC_LIB)
   add_library(FuegoEngine SHARED ${SOURCES} ${PROJECT_HEADERS})
-
 else()
   add_library(FuegoEngine STATIC ${SOURCES} ${PROJECT_HEADERS})
 
+  message(FATAL_ERROR "Target platform is not supported - Compilation denied")
```
Comment (LtSnail) [inline Engine/CMakeLists.txt:85]: I don't understand this change

File: Engine/Fuego/MouseCodes.h [hunk 4]
```diff
@@ -1,5 +1,7 @@
 #pragma once
 
+#include "fupch.h"
```
Comment (LtSnail) [inline Engine/Fuego/MouseCodes.h:3]: We should avoid including pch in headers

File: Engine/Fuego/MouseCodes.h [hunk 5]
```diff
@@ -1,5 +1,7 @@
...
+
```
Comment (LtSnail) [inline Engine/Fuego/MouseCodes.h:4]: Changes in this file are unrelated to the PR

File: Engine/Glad.cmake [hunk 6]
```diff
@@ -0,0 +1,14 @@
+# Glad
```
Comment (LtSnail) [inline Engine/Glad.cmake:1]: I think that here is okay to have a .cmake as glad is a dependency, but you can also convert it to CMakeLists.txt as it is a default way

File: Engine/Glad.cmake [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+set(SOURCES External/glad/src/glad.c)
+
+add_library(Glad STATIC ${SOURCES} ${PROJECT_HEADERS})
+
+target_include_directories(Glad PRIVATE External/glad/include)
+
+set_target_properties(Glad PROPERTIES C_STANDARD 11 C_STANDARD_REQUIRED YES)
+
+target_link_libraries(Glad PRIVATE opengl32)
```
Comment (LtSnail) [inline Engine/Glad.cmake:14]: Could you please double check that we need to link it?

File: Engine/Glad.cmake (hunk 6 above)
Comment (LtSnail) [reply Engine/Glad.cmake:1]: And it also could mess with PRIVATE/PUBLIC

File: Engine/WindowsDep.cmake [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    Fuego/Windows/WindowWin.cpp Fuego/Windows/OpenGL/BufferOpenGL.cpp
+    Fuego/Windows/OpenGL/OpenGLContext.cpp)
+
+set(DEPENDENCIES Glad)
+
+add_library(WindowsDep STATIC ${PROJECT_SOURCES} ${PROJECT_HEADERS})
+
+target_compile_features(WindowsDep PRIVATE cxx_std_20)
+
+target_compile_definitions(WindowsDep PUBLIC FUEGO_PLATFORM_WIN)
```
Comment (LtSnail) [inline Engine/WindowsDep.cmake:21]: These 3 should be in a cross-platform part

File: Engine/WindowsDep.cmake [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+target_compile_definitions(WindowsDep PRIVATE FUEGO_BUILD_LIB)
+target_compile_definitions(WindowsDep PUBLIC FUEGO_ENABLE_ASSERTS)
+
+target_compile_options(WindowsDep PUBLIC /utf-8)
+
+target_include_directories(
+  WindowsDep
+  PUBLIC External/glad/include Fuego/ Fuego/Events ${CMAKE_CURRENT_SOURCE_DIR}
```
Comment (LtSnail) [inline Engine/WindowsDep.cmake:29]: spdlog, Fuego, Fuego/Events should be in a cross-platform part

**Review by LtSnail — CHANGES_REQUESTED** [review]

File: Engine/Glad.cmake (hunk 7 above)
Comment (Mon1Amour4) [reply Engine/Glad.cmake:14]: Yes, we have to

File: Engine/WindowsDep.cmake (hunk 9 above)
Comment (Mon1Amour4) [reply Engine/WindowsDep.cmake:29]: Discussed

**Review by LtSnail — APPROVED** [review]
